assets/src/edit-story/components/library/panes/text/textSets/textSets.js

```diff
@@ -84,25 +90,40 @@ function TextSets() {
       ?.selectedCategory
   );
 
+  const storyPages = useStory(({ state: { pages } }) => pages);
+
   const ref = useRef();
+  const inUseFonts = useRef([]);
 
   const allTextSets = useMemo(() => Object.values(textSets).flat(), [textSets]);
   const categories = useMemo(
-    () =>
-      Object.keys(textSets).map((cat) => ({
+    () => [
+      ...Object.keys(textSets).map((cat) => ({
         id: cat,
         label: CATEGORIES[cat] ?? cat,
       })),
+      { id: 'inUse', label: CATEGORIES.inUse },
+    ],
     [textSets]
   );
 
-  const textSetReducer = useMemo(
-    () => getTextSetReducer(textSets, allTextSets),
-    [textSets, allTextSets]
-  );
+  const getTextSetsForInUseFonts = useCallback(() => {
+    const updatedInUseFonts = getInUseFontsForPages(storyPages);
+    if (
+      updatedInUseFonts.length === inUseFonts.current.length &&
+      updatedInUseFonts.every((font, idx) => font === inUseFonts.current[idx])
```

Comment:
This is the check to ensure that while the array might have changed, let's make sure the values (string font names) are the same.